pyro/contrib/bnn/hidden_layer.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import absolute_import, division, print_function
+
+
+import torch
+from torch.distributions.utils import lazy_property
+import torch.nn.functional as F
+
+from pyro.contrib.bnn.utils import adjoin_ones_vector
+from pyro.distributions.torch_distribution import TorchDistribution
+
+
+class HiddenLayer(TorchDistribution):
+    r"""
+    This distribution is a basic building block in a Bayesian neural network. It represents
```

Comment:
nit: comments wider than 80chars print poorly in jupyter notebooks. feel free to disregard this nit.

pro tip: in vim position cursor at top of paragraph and type `gq<ENTER>` to automatically wrap.